### PR TITLE
example: Add TCP header modification example

### PR DESCRIPTION
**Pull request by ThisSeanZhang**
Old PR is: [#767](https://github.com/libbpf/libbpf-rs/pull/767)

- Replaced tabs with spaces for consistency
- Removed unstable features
- Modified handling of attachment result
- Replaced verbose options and included trigger command in README.md

Thank you for considering a contribution!

In order to streamline review experience for contributors and reviewers, please
be sure to *read* and *follow* the [Contributor's Guide][cont-guide]. It
lays out basic best practices, which, if followed will reduce unnecessary back
and forth and, ultimately, minimize the time it takes to get your change into
the library.

[cont-guide]: https://github.com/libbpf/libbpf-rs/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by ThisSeanZhang** [comment]
How can I install a plugin in VSCode or check the same rules in CI in some other way? :P

**Comment by danielocfb** [comment]
> How can I install a plugin in VSCode or check the same rules in CI in some other way? :P

You don't need VSCode, you can just do what the [workflow](https://github.com/libbpf/libbpf-rs/blob/master/.github/workflows/test.yml) does. So in your case check the clippy and/or rustfmt jobs:
https://github.com/libbpf/libbpf-rs/blob/b47f48eafad0056aa213f1c3df58c22acacfbe45/.github/workflows/test.yml#L185-L205

File: examples/tcp_option/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+            SO_ATTACH_BPF,
+            &prog_fd as *const _ as *const c_void,
+            size_of_val(&prog_fd) as socklen_t,
+        )
+    } {
+        0 => {
+            println!("BPF Attached Successfully!");
+        }
+        _ => {
+            println!("Failed to Attach BPF, Reason: {:?}", Error::last_os_error());
```
Comment (danielocfb) [inline examples/tcp_option/src/main.rs:139]: Should we exit here or is there a good reason for continuing as if nothing happened?

**Review by danielocfb — COMMENTED** [review]
Looks good to me now, thanks! Why did you remove the example from the workspace, though? I think it should stay there.

File: examples/tcp_option/src/main.rs (hunk 1 above)
Comment (ThisSeanZhang) [reply examples/tcp_option/src/main.rs:139]: Should I use panic or return Err or do you have a better suggestion?
```rust
panic!("Failed to Attach BPF, Reason: {:?}", Error::last_os_error());
// or
println!("Failed to Attach BPF, Reason: ");
return Err(Error::last_os_error().into());
```

**Comment by ThisSeanZhang** [comment]
> cargo clippy --locked --no-deps --all-targets --tests -- -D warnings -D clippy::absolute_paths 

I executed cmd and then got this

```test
$ cargo clippy --locked --no-deps --all-targets --tests -- -D warnings -D clippy::absolute_paths 
   Compiling libbpf-rs v0.23.0 (/home/sean/libbpf-rs/libbpf-rs)
error: transmute used without annotations
   --> libbpf-rs/src/print.rs:127:28
    |
127 |         unsafe { Some(mem::transmute(outer_print_cb as *const ())) };
    |                            ^^^^^^^^^ help: consider adding missing annotations: `transmute::<*const (), unsafe extern "C" fn(u32, *const i8, *mut libbpf_sys::__va_list_tag) -> i32>`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#missing_transmute_annotations
    = note: `-D clippy::missing-transmute-annotations` implied by `-D warnings`
    = help: to override `-D warnings` add `#[allow(clippy::missing_transmute_annotations)]`

error: could not compile `libbpf-rs` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error: could not compile `libbpf-rs` (lib) due to 1 previous error
error: could not compile `libbpf-rs` (lib test) due to 1 previous error
```

did I miss something...

File: examples/tcp_option/src/main.rs (hunk 1 above)
Comment (d-e-s-o) [reply examples/tcp_option/src/main.rs:139]: Return is probably best. Thanks!

**Comment by d-e-s-o** [comment]
> > cargo clippy --locked --no-deps --all-targets --tests -- -D warnings -D clippy::absolute_paths
> 
> I executed cmd and then got this
> 
> ```
> $ cargo clippy --locked --no-deps --all-targets --tests -- -D warnings -D clippy::absolute_paths 
>    Compiling libbpf-rs v0.23.0 (/home/sean/libbpf-rs/libbpf-rs)
> error: transmute used without annotations
>    --> libbpf-rs/src/print.rs:127:28
>     |
> 127 |         unsafe { Some(mem::transmute(outer_print_cb as *const ())) };
>     |                            ^^^^^^^^^ help: consider adding missing annotations: `transmute::<*const (), unsafe extern "C" fn(u32, *const i8, *mut libbpf_sys::__va_list_tag) -> i32>`
>     |
>     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#missing_transmute_annotations
>     = note: `-D clippy::missing-transmute-annotations` implied by `-D warnings`
>     = help: to override `-D warnings` add `#[allow(clippy::missing_transmute_annotations)]`
> 
> error: could not compile `libbpf-rs` (lib) due to 1 previous error
> warning: build failed, waiting for other jobs to finish...
> error: could not compile `libbpf-rs` (lib) due to 1 previous error
> error: could not compile `libbpf-rs` (lib test) due to 1 previous error
> ```
> 
> did I miss something...

You may be using nightly `clippy` or a newer version that has added this lint by default. Just allow it for now and move on?

> cargo clippy --locked --no-deps --all-targets --tests -- -D warnings -D clippy::absolute_paths -A clippy::missing-transmute-annotations

**Review by danielocfb — APPROVED** [review]
Looks good to me, thanks again!

**Comment by ThisSeanZhang** [comment]
Thank you very much for your help in this. 🥳

**Pull request merged**